packages/devkit/src/generators/plugin-migrations/aggregate-log-util.ts

```diff
@@ -0,0 +1,60 @@
+import * as chalk from 'chalk';
+import { logger } from 'nx/src/devkit-exports';
+
+interface AggregateLogOptions {
+  project: string;
+  log: string;
+  executorName: string;
+}
+
+interface AggregateLogItem {
+  log: string;
+  projects: Set<string>;
+}
+
+class AggregatedLog {
+  static logs: Map<string, Map<string, AggregateLogItem>> = new Map();
```

Comment:
I'm a bit confused why any of these are static :eyes: 

This makes it hard to separate out different aggregate logs into different instances. Can this all be not static?

Even if that is not what we intend to do right now, I don't really see a point to making that harder. We could instantiate one in the module scope of the migrate utilities which would do the same thing right?

Maybe it would make more sense if I saw a usage of it.

Reply:
It's intentionally static. 

Some `convert-to-inferred` generators will run migration util multiple times for different executors. Static allows us to collect the data and store it in one place. Then we can flush it all at the end of the generator.

If we added a new instance per migration util, we would have collect all that data again for flushing at the end of a single generator.

When we run multiple `convert-to-inferred` generators at once, having multiple instances would compound this. 
Static means that even when running multiple `convert-to-inferred` generators, we have a single state for all the generators, all the executors that are being migrated, and can flush it all, one time, at the end, after all generators have completed.